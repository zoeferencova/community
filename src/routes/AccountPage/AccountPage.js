import React, { Component } from "react";
import { Link } from "react-router-dom";
import CommUnityContext from "../../contexts/context";
import styles from "./AccountPage.module.css"

export default class AccountPage extends Component {
    static contextType = CommUnityContext;

    state = {
        first_name: this.context.currentUser.first_name,
        email: this.context.currentUser.email
    }

    handleChangeName = e => {
        this.setState({ first_name: e.target.value })
    }

    handleChangeEmail = e => {
        this.setState({ email: e.target.value })
    }

    render() {
        return (   
            <main>
                <h3>Account</h3>
                <form className={styles.form}>
                    <div>
                        <label htmlFor="first_name">First Name</label>
                        <input type="text" name="first_name" id="first_name" value={this.state.first_name} onChange={this.handleChangeName} />
                    </div>
                    <div>
                        <label htmlFor="email">Email</label>
                        <input type="email" name="email" id="email" value={this.state.email} onChange={this.handleChangeEmail} />
                    </div>
                    <div>
                        <a href="#">Change password</a>
                        <a href="#">Deactivate account</a>
                    </div>
                    <Link to="/home"><button>Cancel</button></Link>
                    <button type="submit">Save</button>
                </form>
            </main>
        )
    }
}