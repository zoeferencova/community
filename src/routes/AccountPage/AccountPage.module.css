.backButton {
    background: transparent;
    border: none;
    font-size: 23px;
    align-self: flex-start;
    margin-left: 20px;
    color: #79838c;
    opacity: 1;
    transition: ease-in-out 500ms;
    display: none;
}

.backButton:hover {
    cursor: pointer;
    opacity: 0.7;
    transition: ease-in-out 500ms;
}

.links {
    margin-top: 40px;
    align-self: left;
    font-size: 15px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.links a {
    text-decoration: none;
    color: #79838c;
    margin: 6px 0px;
}

.links svg {
    margin-right: 6px;
}

.profPic {
    height: 75px;
    width: 75px;
    border-radius: 80px;
    font-size: 30px;
    margin: 10px auto 40px auto;
    opacity: 0.8;
    font-weight: 500;
}

.detailsContainer {
    display: flex;
    align-items: center;
    width: 100%;
}

.inputContainer {
    display: flex;
    flex-direction: column;
}

.form {
    width: 85%;
}

.buttonContainer {
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin-top: 35px;
}

@media only screen and (min-width: 830px) {
    .form {
        width: 60%;
    }

    .backButton {
        display: block;
    }

}