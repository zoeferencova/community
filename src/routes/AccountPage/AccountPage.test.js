import React from "react";
import { render } from "enzyme";
import AccountPage from "./AccountPage";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

import mockContext from "../../contexts/mock-context";

describe("AccountPage", () => {
    it("renders AccountPage component given mock context data", () => {
        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <BrowserRouter>
                    <AccountPage />
                </BrowserRouter>
            </CommUnityContext.Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});