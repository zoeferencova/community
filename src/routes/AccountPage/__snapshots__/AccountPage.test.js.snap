// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPage renders AccountPage component given mock context data 1`] = `
<main
  class="main"
>
  <button
    class="backButton"
    type="button"
  >
    <i
      class="fas fa-arrow-left"
    />
  </button>
  <form
    class="form"
  >
    <div
      class="circle blue profPic"
      first_name="Zoe"
    >
      Z
    </div>
    <label
      class="label "
      for="first_name"
    >
      First Name
    </label>
    <input
      class="input "
      id="first_name"
      name="first_name"
      required=""
      type="text"
      value="Zoe"
    />
    <label
      class="label "
      for="email"
    >
      Email
    </label>
    <input
      class="input "
      id="email"
      name="email"
      required=""
      type="email"
      value="zoe@gmail.com"
    />
    <div
      class="buttonContainer"
    >
      <button
        class="button logoutButton"
        type="button"
      >
        Log out
      </button>
      <button
        class="buttonDark "
        loading="false"
        type="submit"
      >
        Update
      </button>
    </div>
    <div
      class="links"
    >
      <a
        href="/change-password"
      >
        <i
          class="fas fa-key"
        />
         Change password
      </a>
      <a
        href="/location"
      >
        <i
          class="fas fa-map-marker-alt"
        />
         Change location
      </a>
      <a
        href="/confirm-deactivation"
      >
        <i
          class="fas fa-ban"
        />
         Deactivate account
      </a>
    </div>
  </form>
</main>
`;
